scripts/globals/crafting.lua

```diff
@@ -195,56 +189,113 @@ xi.crafting.unionRepresentativeEventUpdateRenounce = function(player, craftID)
 end
 
 xi.crafting.unionRepresentativeTriggerRenounceCheck = function(player, eventId, realSkill, rankCap, param3)
-    if player:getLocalVar("renounceDialog") == 0 then
+    if player:getLocalVar("skipRenounceDialog") == 0 then
         local count   = 0
         local bitmask = 0
 
         for craftID = xi.skill.WOODWORKING, xi.skill.COOKING do
             local rank = player:getSkillRank(craftID)
-            if rank < 7 then
-                bitmask = bit.bor(bitmask, bit.lshift(1, craftID - 48))
-            else
+
+            if rank >= xi.crafting.rank.ARTISAN then
                 count = count + 1
+            else
+                -- This needs checking. Probably made-up.
+                bitmask = bit.bor(bitmask, bit.lshift(1, craftID - 48))
             end
         end
 
-        if count > 1 then
-            player:setLocalVar("renounceDialog", 1)
+        if count >= 2 then
+            player:setLocalVar("skipRenounceDialog", 1)
+
+            -- TODO: Param 3 is taken directly from captures, but now we know this is incorrect.
+            -- TODO: bitmask needs checking.
             player:startEvent(eventId, VanadielTime(), realSkill, rankCap, param3, 0, 0, count, bitmask)
+
             return true
         end
     end
 
     return false
 end
 
+--------------------------------------------------
+-- Guild Point NPCs (Union Representatives)
+--------------------------------------------------
+xi.crafting.unionRepresentativeTrade = function(player, npc, trade, csid, guildID)
```

Comment:
Outside of scope, but can we possibly nuke calling it unionRepresentative?  That's vague as heck, and even as a decades-long player, I'd have no clue.

Reply:
Certanly. I was thinking
guildPointNPC -> For the NPC that handles guild points and rewards
guildMasterNPC -> For the NPC that handles rank-ups, rank renouncement and expert quests
guildSupportNPC -> for the (paid) synthesis support npc